BackofficeBundle/Resources/public/coffee/orchestraLib.coffee

```diff
@@ -151,7 +151,72 @@ activateSelect2 = (element) ->
       container.parent().addClass('bg-color-red').attr('style', 'border-color:#a90329!important') if term.isNew
       term.text
   )
+#TOKENINPUT ENABLED
+activateToken = (element) ->
+  testLuceneExpression = (obj) ->
+    testedString = obj.val()
+    getBalancedBracketsRegExp = new RegExp(/\([^\(\)]*\)/)
+    getBracketRegExp = new RegExp(/(\(|\))/)
+    getAdjacentOperatorsRegExp = new RegExp(/(\+ \+|- -|\+ -|- \+|\(\))/)
+    isEndingWithOperatorRegExp = new RegExp(/(\+|-)$/)
+    isLuceneExpression = !getAdjacentOperatorsRegExp.test(testedString) and !isEndingWithOperatorRegExp.test(testedString)
+    while isLuceneExpression and getBalancedBracketsRegExp.test(testedString)
```

Comment:
While is dangerous, on some case it could lead to an infinite loop. Is there another way to do this ?